functions/src/roles/roles.ts

```diff
@@ -0,0 +1,60 @@
+import { firestore } from "../admin/admin";
+const admin = require('firebase-admin');
+
+/**
+ * Data object = ['docName string', 'permission string']
+ */
+
+async function createRole(request: any, response: any) {
+    const data = JSON.parse(request.body);
+    
+    const docName = data[0]
+    const permission = data[1]
+    try{
+      await firestore.collection("roles").doc(docName).set({
+        permissions: admin.firestore.FieldValue.arrayUnion(permission) 
+      });
+      response.json(data)
+    } catch(error){
+      response.json("you broke it")
+    }
+  };
+
+  /**
+ * Data object = ['docName string', 'permission string']
+ */
+
+async function updateRole(request: any, response: any)  {
+    const data = JSON.parse(request.body);
+    
+// updates a new document in collection "roles"
+    const docName = data[0]
+    const permission = data[1]
+    try{
+      await firestore.collection("roles").doc(docName).update({
+        permissions: admin.firestore.FieldValue.arrayUnion(permission) 
+      });
+      response.json(data)
+    } catch(error){
+      response.json("you broke it")
```

Comment:
maybe instead of you broke it, you could add a more specific message relating to the cloud function